tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -48,14 +49,37 @@
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, local=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if local:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except:
+                run_id = roslaunch.core.generate_run_id()
```

Comment:
Yeah, that code from `roslaunch` has the same deficiency. I'd be happy with a message.

One thing that brings up is that the roslaunch snippet you referenced busy waits on the master, why shouldn't this code do the same? Is it because that should never happen? And if that's the case why catch and silently pass at all? It's not clear to me what might happen and what this options should do in each of those cases.